### PR TITLE
bug-fix/internal-issue-#604

### PR DESCRIPTION
**Pull request by gnusi**
https://github.com/arangodb/backlog/issues/604

### CODE REVIEW

**Comment by gnusi** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/4993/

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4998/

File: arangod/RestServer/FlushFeature.cpp [hunk 1]
```diff
@@ -48,43 +48,62 @@
...
+      std::string const& type, // subscription type
+      TRI_voc_tick_t databaseId, // vocbase id
+      arangodb::StorageEngine const& engine // vocbase engine
+  ): _databaseId(databaseId),
+     _engine(engine),
+     _tickCurrent(0), // default (smallest) tick for StorageEngine
+     _tickPrevious(0), // default (smallest) tick for StorageEngine
+     _type(type) {
+    _tickCurrent = _engine.currentTick();
+    resetCurrentTick(_tickCurrent);
```
Comment (goedderz) [inline arangod/RestServer/FlushFeature.cpp]: The second line sets `_tickPrevious = _tickCurrent`. Is that really intended? Looks to me we cannot be sure that these are already scanned.

File: arangod/RestServer/FlushFeature.cpp [hunk 2]
```diff
@@ -48,43 +48,62 @@
...
+  }
+
+  void resetCurrentTick(TRI_voc_tick_t tick = 0) noexcept {
```
Comment (goedderz) [inline arangod/RestServer/FlushFeature.cpp]: The default `= 0` here is never used. As far as I understand it, the ticks should be monotonically increasing, so the default doesn't make much sense, either; I'd vote to remove it.
Also, maybe an assertion on `tick >= _tickCurrent` wouldn't hurt?

File: arangod/RestServer/FlushFeature.cpp [hunk 3]
```diff
@@ -48,43 +48,62 @@
...
+    _tickPrevious = _tickCurrent;
+    _tickCurrent = tick;
+  }
+
+  virtual Result commitImpl(VPackSlice data) = 0;
+
+  TRI_voc_tick_t const _databaseId;
+  arangodb::StorageEngine const& _engine;
+  TRI_voc_tick_t _tickCurrent; // last successful tick, should be replayed
+  TRI_voc_tick_t _tickPrevious; // previous successful tick, should be replayed
```
Comment (goedderz) [inline arangod/RestServer/FlushFeature.cpp]: I thought the replay was definitely done up to `_previousTick`?

**Comment by gnusi** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/5006/

**Pull request merged**